Реализована обработка команд: /start, /cat, /dog, /shelter_info, /shedule, /safety, /help, /pass, /back, /stop

koroliana opened the pull request:
Настройка окружения. Приветствие с выбором приюта. Вывод команд 1 этапа в чат. Расписание работы приюта кошек/собак. Выдача рекомендаций по технике безопасности. Обращение к волонтеру. Получение контактных данных для оформления пропуска.
Выполнены задачи: #1, #2.1, #2.2, #3, #10, #5, #6, #8, #10

koroliana committed:
Подготовила описание для кошачьего приюта

koroliana committed:
Подготовила описание для кошачьего приюта

Pruglo92 committed:
#1 Настройка окружения : редактирование файлов application.yml и pom.xml, добавление конфигураций liquibase.

Pruglo92 committed:
Merge pull request #1 from Microd18/#1

#1 Настройка окружения : редактирование файлов application.yml и pom.…

Pruglo92 committed:
#2.1 Реализовать приветствие с выбором приюта для кошек или собак

Pruglo92 committed:
Merge pull request #2 from Microd18/#2.1

#2.1 Реализовать приветствие с выбором приюта для кошек или собак

Pruglo92 committed:
#2.2 Реализовать приветствие с выбором приюта для кошек или собак (добавил команды для 1 этапа в enum)

Pruglo92 committed:
Merge pull request #3 from Microd18/#2.2

#2.2 Реализовать приветствие с выбором приюта для кошек или собак (до…

Pruglo92 committed:
#2.2 Реализация приветствия с выбором приюта для кошек или собак и с выбором информации о приюте, вывод команд 1 этапа в чат

Pruglo92 committed:
Merge pull request #4 from Microd18/#2.2

#2.2 Реализация приветствия с выбором приюта для кошек или собак и с …

Pruglo92 committed:
#2.2 Реализация приветствия с выбором приюта для кошек или собак (FIX : переработка логики хранения состояний)

Pruglo92 committed:
Merge pull request #5 from Microd18/#2.2

#2.2 Реализация приветствия с выбором приюта для кошек или собак (FIX…

DmitriiVolkovIzh committed:
#3 Реализовано расписание работы приюта кошек/собак

#6 Реализовано выдача рекомендаций техники безопасности на территории приюта

#8 Реализовано обращение к волонтеру, если бот не может ответить на вопросы клиента

DmitriiVolkovIzh committed:
Merge pull request #7 from Microd18/#3

#3 Реализовано расписание работы приюта кошек/собак

#6 Реализовано выдача рекомендаций техники безопасности на территории приюта

#8 Реализовано обращение к волонтеру, если бот не может ответить на вопросы клиента

Microd18 committed:
Реализовано получение контактных данных для получения пропуска

Microd18 committed:
Merge pull request #8 from Microd18/Microd18#5

Реализовано получение контактных данных для получения пропуска

Pruglo92 committed:
#2.2 Реализация приветствия с выбором приюта для кошек или собак (FIX : обработка /shelter_info в состоянии SHELTER_INFO)

Pruglo92 committed:
#2.2 Реализация приветствия с выбором приюта для кошек или собак (FIX : обработка /shelter_info в состоянии SHELTER_INFO)

Pruglo92 committed:
Merge pull request #9 from Microd18/#2.2

#2.2 Реализация приветствия с выбором приюта для кошек или собак (FIX…

Pruglo92 committed:
#10 Добавить JavaDoc - добавил JavaDoc для класса, метода, поля.

Pruglo92 committed:
#10 Добавить JavaDoc - добавил JavaDoc для класса, метода, поля.

Pruglo92 committed:
#10 Добавить JavaDoc - добавил JavaDoc для класса, метода, поля.

Pruglo92 committed:
Merge remote-tracking branch 'origin/#10' into #10

# Conflicts:
#	src/main/java/pro/sky/AnimalShelter/enums/BotCommand.java

Pruglo92 committed:
Merge pull request #11 from Microd18/#10

#10

Pruglo92 committed:
Merge pull request #13 from Microd18/#10

#10

DmitriiVolkovIzh committed:
#8.1 поправил обращение к волонтеру

DmitriiVolkovIzh committed:
#8.1 поправил обращение к волонтеру

koroliana committed:
Merge pull request #15 from Microd18/develop

Release 1

dmitry-bizin commented on src/main/java/pro/sky/AnimalShelter/state/ChatStateHolder.java:
```diff
@@ -0,0 +1,94 @@
+    /**
+     * Мапа для отслеживания состояния запущенности бота в каждом чате.
+     */
+    private final Map<Long, Boolean> isBotStartedMap = new HashMap<>();
```
обычно методы, которые возвращают boolean, называют с is. Поля и переменные лучше называть без is

dmitry-bizin replied:
состояние лучше хранить в БД, чтобы оно у вас сохранялось между перезапусками бота

dmitry-bizin commented on src/main/java/pro/sky/AnimalShelter/state/ChatStateHolder.java:
```diff
@@ -0,0 +1,94 @@
+    /**
+     * Мапа для хранения истории состояний чатов.
+     */
+    private final Map<Long, Deque<BotCommand>> chatStateHistory = new HashMap<>();
```
состояние лучше хранить в БД, чтобы оно у вас сохранялось между перезапусками бота

koroliana committed:
Смерджила подключение БД

koroliana committed:
Создала репозитории и сервисы к ним. Полностью заменила работу с ChatStateHolder на ChatStateService
Пока не запускается.

koroliana committed:
Подключение БД

koroliana committed:
Добавила liquibase changeset, но пока ошибка с типом колонки ENUM
Временно закомментила liquibase в pom.xml

koroliana committed:
Добавила в БД enum теперь падает из-за того, что тип "bot_command" уже существует (удалила из истории миграцию)

koroliana committed:
Добавила в БД значения, которые может принимать собственный ENUM bot_command
Падает с ошибкой wrong column type encountered in column [current_state] in table [chat_state]; found [bot_command (Types#VARCHAR)], but expecting [int4 (Types#INTEGER)]

koroliana committed:
корректировка в rollback

koroliana committed:
Подключение БД

koroliana committed:
возврат токена, форматирование кода

Microd18 committed:
Изменил команду /pass, теперь для собак/кошек, выводится разная информация

koroliana committed:
Подключение БД

koroliana committed:
Ошибка: нарушаются ограничения уникальности в БД
- временно primary keys заменены на unique в миграциях
- обработаны ошибки в handlers, связанные с состояниями
- убран тип enum из бд, реализована аннотация @Enumerated(EnumType.STRING) в entity
- закомментированы связи между таблицами, прописанные в entity
- токен убран в переменную окружения

Microd18 committed:
Merge remote-tracking branch 'origin/develop' into Microd18#5

# Conflicts:
#	src/main/java/pro/sky/AnimalShelter/enums/BotCommand.java
#	src/main/java/pro/sky/AnimalShelter/handlers/PassCommandHandler.java

Pruglo92 committed:
#23 Добавить реализацию меню: Как взять животное из приюта (этап 2) (/adopt) + добавление команд для 2 меню + REFACTOR(вынесение общих методов в утилитарный класс)

Pruglo92 committed:
Merge branch 'develop' into #23

# Conflicts:
#	src/main/java/pro/sky/AnimalShelter/enums/BotCommand.java

Pruglo92 committed:
Merge pull request #20 from Microd18/#23

#23 Добавить реализацию меню: Как взять животное из приюта (этап 2) (…

Pruglo92 committed:
#23 Добавить реализацию меню: Как взять животное из приюта (этап 2)FIX

Pruglo92 committed:
Merge pull request #21 from Microd18/#23

#23 Добавить реализацию меню: Как взять животное из приюта (этап 2)FIX

Pruglo92 committed:
#13 Разработать имена пакетов для каждого меню бота и сложить в них все обработчики

Pruglo92 committed:
Merge pull request #22 from Microd18/#13

#13 Разработать имена пакетов для каждого меню бота и сложить в них в…

Pruglo92 committed:
#6666 Подключение бд

DmitriiVolkovIzh committed:
#14 Реализовано правила знакомства с животным DatingRulesHandler

koroliana committed:
Подружила БД и приложение
Сохраняю состояние в БД, и хожу с помощью него по меню
Добавила дополнительную миграцию для переименования переменной в таблице chat_state
Закомментировала логику isBotStarted в StartCommandHandler

koroliana committed:
Merge remote-tracking branch 'origin/koroliana#11' into koroliana#11

# Conflicts:
#	src/main/java/pro/sky/AnimalShelter/entity/ChatState.java
#	src/main/java/pro/sky/AnimalShelter/entity/User.java
#	src/main/java/pro/sky/AnimalShelter/handlers/BackCommandHandler.java
#	src/main/java/pro/sky/AnimalShelter/handlers/HelpCommandHandler.java
#	src/main/java/pro/sky/AnimalShelter/handlers/PassCommandHandler.java
#	src/main/java/pro/sky/AnimalShelter/handlers/SafetyCommandHandler.java
#	src/main/java/pro/sky/AnimalShelter/handlers/ScheduleCommandHandler.java
#	src/main/java/pro/sky/AnimalShelter/handlers/ShelterCommandHandler.java
#	src/main/java/pro/sky/AnimalShelter/handlers/ShelterInfoCommandHandler.java
#	src/main/java/pro/sky/AnimalShelter/handlers/generalHandlers/StartCommandHandler.java
#	src/main/java/pro/sky/AnimalShelter/handlers/generalHandlers/StopCommandHandler.java
#	src/main/java/pro/sky/AnimalShelter/repository/UserRepository.java
#	src/main/java/pro/sky/AnimalShelter/service/ChatStateService.java
#	src/main/java/pro/sky/AnimalShelter/service/UserService.java
#	src/main/resources/db.changelog/changes/db.changelog-1.0.sql

koroliana committed:
Подружила БД и приложение
Сохраняю состояние в БД, и хожу с помощью него по меню
Добавила дополнительную миграцию для переименования переменной в таблице chat_state
Закомментировала логику isBotStarted в StartCommandHandler

DmitriiVolkovIzh committed:
Merge pull request #23 from Microd18/#14

 #14 Реализовано правила знакомства с животным DatingRulesHandler

koroliana committed:
Изменена логика isBotStarted. Теперь она работает.
В логику stopBot добавлена очистка всех состояний
В StartCommandHandler раскомментировала проверку на то, запущен ли бот.

koroliana committed:
Задача#11 Реализация работы бота с сохранением состояний в БД

Koroliana#11

Pruglo92 committed:
#11HotFiX исправление багов после подключения бд, оптимизация кода

Pruglo92 committed:
Merge pull request #25 from Microd18/#11HotFix

#11HotFiX исправление багов после подключения бд, оптимизация кода

DmitriiVolkovIzh committed:
# 15 Реализовано выдачу списка документов, необходимых для того, чтобы взять животное из приюта DocumentsListHandler

DmitriiVolkovIzh committed:
Merge pull request #26 from Microd18/#14

 # 15 Реализовано выдачу списка документов, необходимых для того, что…

DmitriiVolkovIzh committed:
# Правка хэдлеров

DmitriiVolkovIzh committed:
Merge pull request #27 from Microd18/#15

 # Правка хэндлеров

DmitriiVolkovIzh committed:
# Поправил DatingRulesHandler и DocumentsListHandler

DmitriiVolkovIzh committed:
Merge pull request #28 from Microd18/#15

 # Поправил DatingRulesHandler и DocumentsListHandler

DmitriiVolkovIzh committed:
Добавил Handler'ы для меню Adopt

DmitriiVolkovIzh committed:
Merge pull request #29 from Microd18/#15

Добавил оставшиеся Handler'ы для меню Adopt

Microd18 committed:
изменил конструкцию if/else для правильной работы команды /pass

Microd18 committed:
Merge pull request #30 from Microd18/Microd18#17

изменил конструкцию if/else для правильной работы команды /pass

Pruglo92 committed:
#12 Добавить в БД таблицу с chatID, объединяющую данные для users и chat_state

Pruglo92 committed:
Merge pull request #31 from Microd18/#12

#12 Добавить в БД таблицу с chatID, объединяющую данные для users и c…

Microd18 committed:
покрыл тестами сервисы

Microd18 committed:
покрыл тестами сервисы

DmitriiVolkovIzh committed:
Добавил Handler для /transportation_recommendation

DmitriiVolkovIzh committed:
Merge pull request #33 from Microd18/#15

Добавил Handler для /transportation_recommendation

Pruglo92 committed:
#7 Реализовать принятие и запись контактных данных для связи + Unit тесты для UserService

Pruglo92 committed:
Merge pull request #34 from Microd18/#7

#7 Реализовать принятие и запись контактных данных для связи + Unit т…

koroliana committed:
Добавила описание приюта кошек и собак

koroliana committed:
Merge pull request #35 from Microd18/koroliana#4

Задача #4 Добавлен DescriptionCommandHandler

Pruglo92 committed:
Merge pull request #32 from Microd18/develop

pull request for homework
